src/test/components/UploadedRecordingsHome.test.js

```diff
@@ -145,4 +148,55 @@ describe('UploadedRecordingsHome', () => {
 
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  it('can rename stored profiles', async () => {
+    // Store a single profile that we will rename. The data inside is mostly
+    // arbirtrary except the "name" field.
+    // mockDate('4 Jul 2020 15:00');
+    await storeProfileData({
+      profileToken: 'FAKE_1234',
+      jwtToken: null,
+      publishedDate: new Date('4 Jul 2020 13:00'), // This is the future!
+      name: 'Initial Profile Name',
+      preset: null,
+      originHostname: 'https://mozilla.org',
+      meta: {
+        product: 'Firefox',
+        platform: 'Macintosh',
+        toolkit: 'cocoa',
+        misc: 'rv:62.0',
+        oscpu: 'Intel Mac OS X 10.12',
+      },
+      urlPath: '/public/FAKE_1234/marker-chart/',
+      publishedRange: { start: 2000, end: 40000 },
+    });
+    const store = blankStore();
+    {
+      const { findByText } = render(
+        <Provider store={store}>
+          <UploadedRecordingsHome />
+        </Provider>
+      );
+
+      expect(await findByText(/Initial Profile Name/)).toBeTruthy();
+    }
+
+    const newUrlState = stateFromLocation({
+      pathname: `/public/FAKE_1234/marker-chart/`,
+      search: '',
+      hash: '',
+    });
+    store.dispatch(updateUrlState(newUrlState));
+
+    await store.dispatch(changeProfileName('My Profile Name'));
```

Comment:
Thanks for the explanation!

To have a better test and easier to maintain, I believe we should have 2 different stores (to simulate 2 different pages):
* store 1 for the list of recordings => a `blankStore` is fine for that one.
* store 2 for the profile viewer => you can use `blankStore` + dispatch the url state change for this one like you already did, and dispatch the profile name change on this store.